### PR TITLE
Update SDK to .NET 6.0.100 and enable nullable reference types

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -30,8 +30,8 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.VisualStudio.Internal.MicroBuild.VisualStudio" Version="$(MicroBuildVersion)" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.11.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="4.0.0-6.final" PrivateAssets="all" />
```
Comment (JoeRobich) [inline Directory.Build.props:34]: There is a '4.0.0' package which is published as unlisted. A 4.0.1 package should be published today or tomorrow depending on validation.

**Review by JoeRobich — APPROVED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (sharwell) [reply Directory.Build.props:34]: The 4.0.1 update should be an easy follow-up once available 👍

Comment (JoeRobich) [reply Directory.Build.props:34]: Sounds good

Comment (JoeRobich) [reply Directory.Build.props:34]: 4.0.1 packages are now available.

Comment (sharwell) [reply Directory.Build.props]: ➡️ Now using 4.0.1

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**